ntoskrnl/ke/i386/cpu.c

```diff
@@ -101,14 +101,15 @@ KiSetProcessorType(VOID)
      * Model and the Step, while the Type contains the returned Type.
      * We ignore the family.
      *
-     * For the stepping, we convert this: zzzzzzxy into this: x0y
+     * For the stepping, we convert this: zzzxzzxy into this: xx0y
      */
     Stepping = CpuInfo.Eax & 0xF0;
     Stepping <<= 4;
-    Stepping += (CpuInfo.Eax & 0xFF);
-    Stepping &= 0xF0F;
+    Stepping += ((CpuInfo.Eax & 0x000F0000) >> 4);
+    Stepping += (CpuInfo.Eax & 0xF);
     Type = CpuInfo.Eax & 0xF00;
     Type >>= 8;
+    Type += ((CpuInfo.Eax & 0x0FF00000) >> 16);
```

Comment:
The top of the three nibbles is going to be ignored since we store this in an UCHAR. I would suggest making that explicit by masking out those bits, as well as adding a comment noting that there are additional family bits but we can't make use of them.

Reply:
Actually the way Windows encodes these seems to be:
If the "family" (0xf00) value is less than 15, use only that value
If it is 15, use "extended family" (0x0ff00000) but add 15 to distinguish it from non-extended values.

Reply:
Mentioning this reference (URL) in either the commit log and/or the comment in source code would be nice :)